### PR TITLE
Detect imports in `src` layouts by default

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

Occasionally, we receive bug reports that imports in `src` directories aren't correctly detected. The root of the problem is that we default to `src = ["."]`, so users have to set `src = ["src"]` explicitly. This PR extends the default to cover _both_ of them: `src = [".", "src"]`.

Closes https://github.com/astral-sh/ruff/issues/12454.

## Test Plan

I replicated the structure described in https://github.com/astral-sh/ruff/issues/12453, and verified that the imports were considered sorted, but that adding `src = ["."]` showed an error.

### CODE REVIEW

**Review by zanieb — APPROVED** [review]
I guess the ecosystem tests will provide some test coverage?

**Comment by charliermarsh** [comment]
Well, I guess it's not "a lot of ecosystem changes" -- it's ecosystem changes in two projects, and for projects that _are_ effected, it's a lot of changes.

**Review by MichaReiser — COMMENTED** [review]
Thanks @charliermarsh 

Do we also need to change 
https://github.com/astral-sh/ruff/blob/25f5a8b2019260eba3889d9800a60950a810a43e/crates/ruff_linter/src/settings/mod.rs#L399

File: crates/ruff_workspace/src/options.rs [hunk 1]
```diff
@@ -341,15 +341,17 @@ pub struct Options {
     ///
     /// When omitted, the `src` directory will typically default to the
     /// directory containing the nearest `pyproject.toml`, `ruff.toml`, or
-    /// `.ruff.toml` file (the "project root"), unless a configuration file
-    /// is explicitly provided (e.g., via the `--config` command-line flag).
+    /// `.ruff.toml` file (the "project root") along with a `"src"` subdirectory,
```
[the remark this answers is not in the document]
Comment (charliermarsh) [reply crates/ruff_workspace/src/options.rs:344]: Yeah I think so.

**Review by AlexWaygood — APPROVED** [review]
LGTM other than @MichaReiser's comments

File: crates/ruff_workspace/src/options.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (charliermarsh) [reply crates/ruff_workspace/src/options.rs]: Done, I just used `lib`

**Comment by NasaGeek** [comment]
Update to FAQ.md at https://github.com/astral-sh/ruff/blob/main/docs/faq.md#:~:text=Depending%20on%20your%20project%20structure is merited as a result of this change I think.

**Pull request merged**